src/prpy/tsr/tsr.py

```diff
@@ -52,36 +71,323 @@ def __init__(self, T0_w=None, Tw_e=None, Bw=None,
+                r_offset = numpy.arctan2(-rot[0, 1], -rot[0, 2])
+                rpy[0] = r_offset
+                rpy[1] = -pi/2
+                rpy[2] = 0.
+        return rpy
+
+    @staticmethod
+    def trans_to_xyzrpy(trans):
+        """
+        Converts a transformation matrix to one valid xyzrpy
+        @param trans 4x4 transformation matrix
+        @return xyzrpy 6x1 xyzrpy
+        """
+        xyz, rot = trans[0:3, 3], trans[0:3, 0:3]
+        rpy = TSR.rot_to_rpy(rot)
+        return numpy.hstack((xyz, rpy))
+
+    @staticmethod
+    def rpy_to_rot(rpy):
+        """
+        Converts an rpy to a rotation matrix
+        @param rpy 3x1 rpy
+        @return rot 3x3 rotation matrix
+        """
+        rot = numpy.zeros((3, 3))
+        r, p, y = rpy[0], rpy[1], rpy[2]
+        rot[0][0] = numpy.cos(p)*numpy.cos(y)
+        rot[1][0] = numpy.cos(p)*numpy.sin(y)
+        rot[2][0] = -numpy.sin(p)
+        rot[0][1] = (numpy.sin(r)*numpy.sin(p)*numpy.cos(y) -
+                     numpy.cos(r)*numpy.sin(y))
+        rot[1][1] = (numpy.sin(r)*numpy.sin(p)*numpy.sin(y) +
+                     numpy.cos(r)*numpy.cos(y))
+        rot[2][1] = numpy.sin(r)*numpy.cos(p)
+        rot[0][2] = (numpy.cos(r)*numpy.sin(p)*numpy.cos(y) +
+                     numpy.sin(r)*numpy.sin(y))
+        rot[1][2] = (numpy.cos(r)*numpy.sin(p)*numpy.sin(y) -
+                     numpy.sin(r)*numpy.cos(y))
+        rot[2][2] = numpy.cos(r)*numpy.cos(p)
+        return rot
+
+    @staticmethod
+    def xyzrpy_to_trans(xyzrpy):
+        """
+        Converts an xyzrpy to a transformation matrix
+        @param xyzrpy 6x1 xyzrpy vector
+        @return trans 4x4 transformation matrix
+        """
+        trans = numpy.zeros((4, 4))
+        trans[3][3] = 1.0
+        xyz, rpy = xyzrpy[0:3], xyzrpy[3:6]
+        trans[0:3, 3] = xyz
+        rot = TSR.rpy_to_rot(rpy)
+        trans[0:3, 0:3] = rot
+        return trans
+
+    @staticmethod
+    def xyz_within_bounds(xyz, Bw):
+        """
+        Checks whether an xyz value is within a given xyz bounds.
+        Main issue: dealing with roundoff issues for zero bounds
+        @param xyz a 3x1 xyz value
+        @param Bw bounds on xyz
+        @return check a 3x1 vector of True if within and False if outside
+        """
+        # Check bounds condition on XYZ component.
+        xyzcheck = [((x + EPSILON) >= Bw[i, 0]) and
+                    ((x - EPSILON) <= Bw[i, 1])
+                    for i, x in enumerate(xyz)]
+        return xyzcheck
+
+    @staticmethod
+    def rpy_within_bounds(rpy, Bw):
+        """
+        Checks whether an rpy value is within a given rpy bounds.
+        Assumes all values in the bounds are [-pi, pi]
+        Two main issues: dealing with roundoff issues for zero bounds and
+        Wraparound for rpy.
+        @param rpy a 3x1 rpy value
+        @param Bw bounds on rpy
+        @return check a 3x1 vector of True if within and False if outside
+        """
```

Comment:
Clarified in https://github.com/personalrobotics/prpy/commit/e49f68ab676c252f9e8340e98a4c9af7e063c9eb.